File: src/main/java/com/procurement/submission/model/dto/response/CheckPeriodResponseDto.java
package com.procurement.submission.model.dto.response;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;

@Getter
@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class CheckPeriodResponseDto {

    private final Boolean isPeriodValid;

    private final Boolean isPeriodChanged;

    @JsonCreator
    public CheckPeriodResponseDto(@JsonProperty("isPeriodValid") final Boolean isPeriodValid,
                                  @JsonProperty("isPeriodChanged") final Boolean isPeriodChanged) {
        this.isPeriodValid = isPeriodValid;
        this.isPeriodChanged = isPeriodChanged;
    }
}

File: src/main/kotlin/com/procurement/submission/model/dto/response/CheckPeriodResponseDto.kt
package com.procurement.submission.model.dto.response

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import java.time.LocalDateTime

@JsonInclude(JsonInclude.Include.NON_EMPTY)
data class CheckPeriodResponseDto @JsonCreator constructor(

        @get:JsonProperty("setExtendedPeriod")
        val setExtendedPeriod: Boolean?,

        @get:JsonProperty("isPeriodChanged")
        val isPeriodChanged: Boolean,

        val tenderPeriod: CheckPeriod
)

data class CheckPeriod @JsonCreator constructor(

        val endDate: LocalDateTime
)
